### ISSUE TITLE
Existential type is a parse error as a class property

### ISSUE THREAD

**Issue by STRML** [facebook/flow#2047]
I've found a pattern similar to the following to be very useful to avoid double-typing:

```js
class Store extends SomeStore {

  data: *;

  initialize(data: InitialData) {
    this.data = {
      foo: data.foo,
      someNumber: 0,
      someBoolean: false
    };
  }
}
```

Flow can correctly parse this & actually throws proper type errors when accessing `this.data`'s members.

However, using the symbol `*` is a parse error:

```
ModuleBuildError: Module build failed: SyntaxError: /project/ui/stores/AnnouncementStore.js: Unexpected token (64:8)
  62 |
  63 |   // eslint-disable-next-line
> 64 |   data: *;
     |         ^
  65 |
  66 |   load(data: InitialData) {
  67 |     this.data = {
  at Parser.pp.raise (/project/node_modules/babylon/lib/parser/location.js:22:13)
  at Parser.pp.unexpected (/project/node_modules/babylon/lib/parser/util.js:89:8)
```

I've tried aliasing the type and some combinations of parentheses, etc, to get it to parse but I can't seem to find the trick.

**Comment by STRML** [comment]
Just for reference, for now I'm doing this:

```
  /*:: data: *; ::*/
```

This works fine.

**Comment by nmn** [comment]
I think this is more an Issue for babel.

**Comment by STRML** [comment]
This was [fixed in Babylon 6.8.1](https://phabricator.babeljs.io/T7365) - closing.
